### ISSUE TITLE
Failed to provision Vivotek IB9371-HT

### ISSUE THREAD

**Issue by jonlindeheim** [aws-quickstart/quickstart-onica-connected-camera#7]
Tried to provision a IB9371-HT which should be supported. Are using FW 0106a which is the latest available from the Vivotek website for that device.

The config-app is sending the following request:

PUT /provisioning/pair HTTP/1.1
host: <my camera ip>
content-length: 4693
Connection: close

With, from what I can see, a correct json-payload. 
Get this response:

HTTP/1.1 404 Not Found
Date: Sat, 01 Jan 2000 00:52:51 GMT
Server: Web Server
Accept-Ranges: bytes
Connection: close
Content-Type: text/html; charset=ISO-8859-1
 
<HTML><HEAD><TITLE>404 Not Found</TITLE></HEAD>
<BODY><H1>404 Not Found</H1>
The requested URL /provisioning/pair was not found on this server.
</BODY></HTML>

**Comment by chrwel** [comment]
Are you running the Vivotek package addon "teleport"? I did not have that at all on my camera and once I deployed it I could stream successfully.

**Comment by swalchek** [comment]
Jonlindeheim,

Download this Vivotek teleport package (https://drive.google.com/open?id=1LKoeRMGhMoIfrhAcEbZhkFqG-rg4jRI8) and follow the instructions in the video. 

Should not take you longer than a couple minutes to get it going!

**Comment by swalchek** [comment]
Here is the link to the video: https://www.dropbox.com/s/l5odvvhxeqr46ll/VIVOTEK_VADP_Package_Update_Instructions.mp4?dl=0

/sw

**Comment by jonlindeheim** [comment]
Thank you! Hard to know that you need some extra application =). Will try it tomorrow!

**Comment by jonlindeheim** [comment]
Got a little bit further now, it gets status "upload" in the config-app. But the camera log has this error:

14:58:24.271|ERROR|22314|KinesisUploader.cpp:224|select socket fail, errno= 4

With no further description. Any ideas how to proceed from here? Maybe turn on some more logging or similar?

**Comment by chrwel** [comment]
I get stuck on the same status "UPLOAD". Any additional insight @swalchek ?

**Comment by jonlindeheim** [comment]
Anyone had any luck with this? @chrwel @swalchek

**Comment by chrwel** [comment]
@jonlindeheim I have not been able to find a solution to it. Hoping we can get a point of view from AWS. @swalchek any insight or comment on the issue below?

**Comment by swalchek** [comment]
Apologies for the delay @jonlindeheim @chrwel, I am on paternity leave at the moment...would it be helpful if I connected you with someone from Vivotek?

**Comment by chrwel** [comment]
@swalchek a Vivotek SME would be very helpful. Can do it over email if you dont want to post his info here.

**Comment by marceloscharan** [comment]
@swalchek where did you find the Teleport package to download?

**Comment by swalchek** [comment]
@marceloscharan reach out David Robo ( drobo@ amazon.com) and he can direct you to the right folks at Vivotek that can help!

**Comment by kevinvogtlowell** [comment]
@jonlindeheim @chrwel @swalchek have any of you had any luck in figuring this out since January? I am also stuck on the Upload status with my Vivotek cam and would really appreciate some guidance!

**Comment by marceloscharan** [comment]
@kevinvogtlowell yeah man the lack of documentation right now is the main problem... But I was able to get working. So here are the things you need to do:

1- Make sure your Vivotek camera has the Teleport package (I'm running version 3.0.0.1), you can see that on the Package Management area on your Vivotek admin panel.

2- Launch the following Cloud Formation template: https://aws-quickstart.s3.amazonaws.com/quickstart-onica-connected-camera/templates/connected-camera-master.template

3- During the template setup you can choose the best region for you and choose to provision key for the camera automatically (if it is your 1st time running this select "yes" here). Also **don't** change anything on the AWS Quick Start Configuration, just leave it the way it is. Run the CFs template (it will take around 5-10min).

4- In Cloud Formation under the main template you just installed go to the Output tab and copy the Stack Endpoint and the Provision Key (if you choose not to automatically provision the key you won't see this value here)

5- Now install the AWS Quick IOT Connector app on your computer (Mac, Windows or Linux). You can download it here: https://github.com/aws-quickstart/quickstart-onica-connected-camera/releases

6- The first time you open the app after the installation it will ask you for the Stack Endpoint and Provisioning Key, please inform the values generated by the Cloud Formation template (step 4)

7- Now make sure you are in the same network as your camera and select "Discover Cameras". After that select your camera and for Vivotek cameras select the HTTP option and specify your username. password (the same you use to get access to your camera's admin page).

8- Click "Provision Selected Cameras". Done!

You should be able to go to Kinesis Video Stream and see your feed there.

If you want a more detailled step y step I would suggest to follow thi AWS guide:
https://aws-quickstart.s3.amazonaws.com/quickstart-onica-connected-camera/doc/aws-iot-camera-connector-on-the-aws-cloud.pdf

I hope I could help you.... Good luck man.

**Comment by kevinvogtlowell** [comment]
@marceloscharan Thanks for the detailed response! Unfortunately, I've done all of that already and followed the quickstarter deployment guide.

My issue occurs during provisioning, after I click "Provision Selected Cameras". I'm able to get the config app to begin provisioning of the cam, but the status of the camera winds up getting stuck on "Upload" without progressing any further or without any indications of errors that can be troubleshot. Has anyone been able to figure out how to get it to finish provisioning?

**Comment by marceloscharan** [comment]
@kevinvogtlowell I honestly don't trust those status... For example right now I have a camera streaming to AWS and the status on the app is "STARTING".

Did you check Kinesis Video Stream to make sure you have a streaming there and see if the video is coming trough? Also did you check your Teleport Package to make sure all the IOT and KVS infos were stored on the package?

**Comment by kevinvogtlowell** [comment]
@marceloscharan you were right! The status was just not changing to paired, but I reinitialized everything and got it to "Upload" status, and I'm now successfully streaming.

Thank you!

**Comment by kd5uwl** [comment]
Hi, I have this exact same issue but on the IB-9365EHT Downloads page I don’t see a “Teleport” application.  The only applications already installed on my camera are IoT Security and VCA.  Where can I find the Teleport application for download, please?

**Comment by lionsheep24** [comment]
I get stuck during adding camera to entry. The error says as follows:
 Unable to determine status : received authentication error from camera. Check camera api username/password.
But the username/password I used was the default username/password, which is vaild to connect camera url.
I can't find out the root cause of this problem.
Please help me to solve this issue.
